src/ConfigPostProcessor.php

```diff
@@ -260,4 +278,35 @@ private function noopReplacement($value)
     {
         return $value;
     }
+
+    private function replaceDependencyFactories(array $config)
+    {
+        $factories = isset($config['factories']) ? $config['factories'] : [];
+        $aliases = isset($config['aliases']) ? $config['aliases'] : [];
```

Comment:
we don't need these variables, in the loop we can just do:

```php
if (!isset($config['aliases'][$service'])) {
    $config['aliases'][$service'] = $replacedService;
}
```

and then we can also remove after the loop:

```php
       if ($aliases !== []) {
             $config['aliases'] = $aliases;
         }
```

the same with `$factories`.

It should reduce complexity and additional variables here.